backend-rust/src/graphql_api.rs

```diff
@@ -4543,22 +4901,116 @@ pub struct ContractResumed {
 }
 
 #[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+#[graphql(complex)]
 pub struct ContractUpdated {
-    contract_address: ContractAddress,
-    instigator:       Address,
-    amount:           Amount,
-    message_as_hex:   String,
-    receive_name:     String,
-    version:          ContractVersion,
-    // eventsAsHex("Returns the first _n_ elements from the list." first: Int "Returns the elements
-    // in the list that come after the specified cursor." after: String "Returns the last _n_
-    // elements from the list." last: Int "Returns the elements in the list that come before the
-    // specified cursor." before: String): StringConnection events("Returns the first _n_
-    // elements from the list." first: Int "Returns the elements in the list that come after the
-    // specified cursor." after: String "Returns the last _n_ elements from the list." last: Int
-    // "Returns the elements in the list that come before the specified cursor." before: String):
-    // StringConnection
-    // TODO message: String,
+    contract_address:  ContractAddress,
+    instigator:        Address,
+    amount:            Amount,
+    receive_name:      String,
+    version:           ContractVersion,
+    // All logged events by the smart contract during the transaction execution.
+    contract_logs_raw: Vec<Vec<u8>>,
+    input_parameter:   Vec<u8>,
+}
+
+#[ComplexObject]
+impl ContractUpdated {
+    async fn message_as_hex(&self) -> ApiResult<String> { Ok(hex::encode(&self.input_parameter)) }
+
+    async fn message<'a>(&self, ctx: &Context<'a>) -> ApiResult<String> {
```

Comment:
```suggestion
    async fn message(&self, ctx: &Context<'_>) -> ApiResult<String> {
```

Reply:
I leave the lifetime name explicitly since it is consistent with the rest of the code currently.